python/ray/tests/test_failure.py

```diff
@@ -291,6 +292,44 @@ def fail_method(self):
     assert error_message1 in errors[0].error_message
 
 
+def test_failed_actor_init_and_restart(ray_start_regular, error_pubsub):
+    p = error_pubsub
+    error_message1 = "actor constructor failed"
+    error_flag_file_name = "/tmp/TEST_ERROR_FLAG_" + str(
+        random.randint(0, 99999))
+
+    @ray.remote(max_restarts=2)
+    class FailedActor:
+        def __init__(self):
+            # This actor will raise error at first time.
+            # Then after restarting, it will be created successfully
+            if not os.path.isfile(error_flag_file_name):
+                open(error_flag_file_name, "w").close()
+                print("flag not exits, raise error")
+                raise Exception(error_message1)
```

Comment:
Use `ray.get_runtime_context().was_current_actor_reconstructed` to simplify this. And could you please also help rename `was_current_actor_reconstructed` to `was_current_actor_restarted`?

Reply:
Done. Renaming was_current_actor_reconstructed is unrelated to this PR, I'll do it in another PR.